mithril-test-lab/mithril-end-to-end/src/mithril/infrastructure.rs

```diff
@@ -280,7 +329,11 @@ impl MithrilInfrastructure {
     }
 
     pub fn build_client(&self) -> StdResult<Client> {
-        Client::new(self.aggregator.endpoint(), &self.work_dir, &self.bin_dir)
+        let era_work_dir = self.work_dir.join(format!("era.{}", self.current_era));
+        if !era_work_dir.exists() {
+            fs::create_dir(&era_work_dir)?;
+        }
```

Comment:
Should we keep previous path if era switch is disabled ?

Reply:
The artifact folder structure is a bit less messy with this new folder IMO, but we could use it only on era switch :+1: